core/src/main/java/com/graphhopper/util/PointList.java

```diff
@@ -449,14 +468,41 @@ public PointList copy(int from, int end) {
         return copyPL;
     }
 
+    /**
+     * Create a shallow copy of this Pointlist from from to end, excluding end.
+     * <p>
+     * This method is unsafe, because you can still change this object. If you do so, you can break the values of the
+     * shallow copy. The shallow copy only contains pointers with from and end and does not know what you change, make
+     * sure to create a new shallow copy after changing this object, if the shallow copy might become corrupt due to your
+     * change.
+     * <p>
+     * If possible, consider using {@link PointList#shallowCopySafe(int, int)}.
+     */
+    public PointList shallowCopyUnsafe(final int from, final int end) {
+        return new ShallowImmutablePointList(from, end, this);
+    }
+
+    /**
+     * Create a shallow copy of this Pointlist from from to end, excluding end.
+     * <p>
+     * This method makes this object immutable, therefore the shallow copy is guaranteed to always contain the correct
+     * references.
+     * <p>
+     * If you know what your are doing, you can also use {@link PointList#shallowCopyUnsafe(int, int)}.
+     */
+    public PointList shallowCopySafe(final int from, final int end) {
```

Comment:
Do we need the safe method? Or maybe use unsafe as a parameter?

Reply:
Would be also possible. I liked the idea of two different methods here, but doesn't really matter and keeps the duplication low. I can merge the two methods.

Reply:
I do not like the name with safe+unsafe (not exactly clear to which word the safe belongs) and shallowCopy (or createShallowCopy?) is nicer